src/Tasks/Microsoft.NET.Build.Tasks/ProcessFrameworkReferences.cs

```diff
@@ -161,8 +156,33 @@ public class ProcessFrameworkReferences : TaskBase
         [Output]
         public string[]? KnownRuntimeIdentifierPlatforms { get; set; }
 
+        /// <summary>
+        /// The runtime identifier used to represent platform-agnostic components - 'any' runtime will suffice!
+        /// </summary>
+        private const string RuntimeIdentifierForPlatformAgnosticComponents = "any";
+
         private Version? _normalizedTargetFrameworkVersion;
 
+        /// <summary>
+        /// Since this Target is mostly focused on managing RID-specific assets, we massage the 'any' RID (which is platform-agnostic) into a 'null'
+        /// value to make processing simpler.
+        /// </summary>
+        private string? EffectiveRuntimeIdentifier => RuntimeIdentifier == RuntimeIdentifierForPlatformAgnosticComponents ? null : RuntimeIdentifier;
+
+        /// <summary>
+        /// If the current project is specifically targeting a platform, as opposed to being platform-agnostic.
+        /// </summary>
+        private bool ProjectIsPlatformSpecific =>
+            !string.IsNullOrEmpty(EffectiveRuntimeIdentifier);
+
+        /// <summary>
+        /// We have several deployment models that require the use of runtime assets of various kinds.
+        /// This member helps identify when any of those models are in use, because we've had bugs in the past
+        /// where we didn't properly account for all of them.
+        /// </summary>
+        private bool DeploymentModelRequiresRuntimeComponents =>
+            SelfContained || ReadyToRunEnabled || PublishAot || RequiresILLinkPack;
```

Comment:
This member is the actual fix. Every other change is tests, infra, debugging utility, clarity, etc.